File: graphes/tools/todot/todot.py
#!/usr/bin/env python
import sys
from pygrap import *

def main():
    """
    main
    """
    if len(sys.argv) <= 2:
        print("Usage: ./todot (in.dot) (out.dot)\n")
        exit(-1)

    pathin = sys.argv[1]
    pathout = sys.argv[2]

    gr = getGraphFromFile(pathin)

    if gr is not None:
        graph_fprint(pathout, gr)

if __name__ == '__main__':
    main()

File: graphs/tools/todot/todot.cpp
#include "todot.hpp"

int main(int argc, char* argv[]){
  if (argc <= 2){
    std::cout << "Usage: ./todot (in.dot) (out.dot)\n";
  }

  std::cerr << "WARNING: ./todot is deprecated, conditions are not exported." << std::endl;
  
  char* pathIn = argv[1];
  char* pathOut = argv[2];

  graph_t* gr;

  FILE* fpIn;
  FILE* fpOut;

  fpIn = fopen(pathIn, "r");
  fpOut = fopen(pathOut, "w");

  if (fpIn == NULL || fpOut == NULL){
    return 1;
  }

  gr = getGraphFromFile(fpIn);
  fclose(fpIn);

  if (gr != NULL){
    graph_fprint(fpOut, gr);
  }
  fclose(fpOut);
}